src/types.ts

```diff
@@ -86,15 +86,66 @@ export type Blockchain =
       ens?: NamingServiceSource;
       zns?: NamingServiceSource;
       cns?: NamingServiceSource;
-      web3Provider?: Web3Provider;
+      web3Provider?: Provider;
+      provider?: Provider;
     };
 
 export type API = {
   url: string;
 };
+type ProviderParams = readonly [
+  (
+    | {
+        [record: string]: string;
+      }
+    | string
+  ),
+  string,
+];
+
+export interface RequestArguments {
+  method: string;
+  params?: unknown[] | object;
+}
+
+export interface Provider {
+  call: (method: string, params: ProviderParams) => Promise<any>;
+}
+
+/**
+ * @deprecated
+ */
+export type Web3Provider = Provider;
+
+export interface JsonRpcPayload {
+  jsonrpc: string;
+  method: string;
+  params: ProviderParams;
+  id?: string | number;
+}
+
+export interface JsonRpcResponse {
+  jsonrpc: string;
+  id: number;
+  result?: any;
+  error?: string;
+}
+
+type ProviderMethod = (
+  payload: JsonRpcPayload,
+  callback: (error: Error | null, result?: JsonRpcResponse) => void,
+) => void;
+
+export interface OldAbstractProvider {
```

Comment:
I'd rename it to something that gives us some context that this is web3 interface. `Web3Version0Provider` maybe or something like that?